src/DonorProfiles/Tabs/TabsRegister.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Give\DonorProfiles\Tabs;
+
+use Give\DonorProfiles\Tabs\Contracts\Tab;
+use http\Exception\InvalidArgumentException;
+
+class TabsRegister {
+	/**
+	 * FQCN of Tab classes
+	 *
+	 * @since 2.11.0
+	 *
+	 * @var string[]
+	 */
+	private $tabs = [];
+
+	/**
+	 * Returns all of the registered tabs
+	 *
+	 * @since 2.11.0
+	 *
+	 * @return string[]
+	 */
+	public function getTabs() {
+		return $this->tabs;
+	}
+
+	/**
+	 * Checks to see if a tab is registered with the given ID
+	 *
+	 * @since 2.11.0
+	 *
+	 * @param string $id
+	 *
+	 * @return bool
+	 */
+	public function hasTab( $id ) {
+		return isset( $this->tabs[ $id ] );
+	}
+
+	/**
+	 * Returns a tab with the given ID
+	 *
+	 * @since 2.11.0
+	 *
+	 * @param string $id
+	 *
+	 * @return string
+	 */
+	public function getTab( $id ) {
+		if ( ! isset( $this->tabs[ $id ] ) ) {
+			throw new \InvalidArgumentException( "No migration exists with the ID {$id}" );
+		}
+
+		return $this->migrations[ $id ];
+	}
+
+	/**
+	 * Returns all of the registered tab ids
+	 *
+	 * @since 2.11.0
+	 *
+	 * @return string[]
+	 */
+	public function getRegisteredIds() {
+		return array_keys( $this->tabs );
+	}
+
+	/**
+	 * Add a tab to the list of tabs
+	 *
+	 * @since 2.11.0
+	 *
+	 * @param string $tabClass FQCN of the Tab Class
+	 */
+	public function addTab( $tabClass ) {
+		if ( ! is_subclass_of( $tabClass, Tab::class ) ) {
+			throw new \InvalidArgumentException( 'Class must extend the ' . Tab::class . ' class' );
+		}
+
+		$tabId = $tabClass::id();
+
+		if ( isset( $this->tabs[ $tabId ] ) ) {
```

Comment:
I would suggest re-using the `hasTab` method here, instead of duplicating the `isset` logic.

Reply:
Good thinking! Done, ref: https://github.com/impress-org/givewp/pull/5577/commits/8b485cfb89418896ff7c91c70fb6574e2c83ae63